src/app/inbound.rs

```diff
@@ -32,6 +33,7 @@ impl From<SocketAddr> for Endpoint {
         Self {
             addr,
             dst_name: None,
+            http_settings: settings::Settings::Http2,
```

Comment:
Would it make more sense to have a distinct Settings variant for "NotHttp"? I think it might be less surprising next time someone comes through this -- it's subtle when/how this is used...  And we could fail to build clients when this settings variant is used.